src/internal/m365/collection/drive/collections_tree_test.go

```diff
@@ -508,3 +472,942 @@ func (suite *CollectionsTreeUnitSuite) TestCollections_GetTree() {
+// and is not concerned with unifying previous paths or post-processing collections
+func (suite *CollectionsTreeUnitSuite) TestCollections_PopulateTree() {
+	drv := models.NewDrive()
+	drv.SetId(ptr.To(id(drive)))
+	drv.SetName(ptr.To(name(drive)))
+
+	type expected struct {
+		counts                   countTD.Expected
+		err                      require.ErrorAssertionFunc
+		treeSize                 int
+		treeContainsFolderIDs    []string
+		treeContainsTombstoneIDs []string
+	}
+
+	table := []struct {
+		name       string
+		enumerator mock.EnumerateItemsDeltaByDrive
+		tree       *folderyMcFolderFace
+		limiter    *pagerLimiter
+		expect     expected
+	}{
+		{
+			name: "nil page",
+			tree: newFolderyMcFolderFace(nil),
+			enumerator: mock.EnumerateItemsDeltaByDrive{
+				DrivePagers: map[string]*mock.DriveItemsDeltaPager{
+					id(drive): {
+						Pages:       nil,
+						DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
+					},
+				},
+			},
+			limiter: newPagerLimiter(control.DefaultOptions()),
+			expect: expected{
+				counts:                   countTD.Expected{},
+				err:                      require.NoError,
+				treeSize:                 0,
+				treeContainsFolderIDs:    []string{},
+				treeContainsTombstoneIDs: []string{},
+			},
+		},
+		{
+			name: "root only",
+			tree: newFolderyMcFolderFace(nil),
+			enumerator: mock.EnumerateItemsDeltaByDrive{
+				DrivePagers: map[string]*mock.DriveItemsDeltaPager{
+					id(drive): {
+						Pages:       pagesOf(rootAnd()),
+						DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
+					},
+				},
+			},
+			limiter: newPagerLimiter(control.DefaultOptions()),
+			expect: expected{
+				counts: countTD.Expected{
+					count.TotalFoldersProcessed: 1,
+					count.PagesEnumerated:       1,
+				},
+				err:      require.NoError,
+				treeSize: 1,
+				treeContainsFolderIDs: []string{
+					rootID,
+				},
+				treeContainsTombstoneIDs: []string{},
+			},
+		},
+		{
+			name: "root only on two pages",
+			tree: newFolderyMcFolderFace(nil),
+			enumerator: mock.EnumerateItemsDeltaByDrive{
+				DrivePagers: map[string]*mock.DriveItemsDeltaPager{
+					id(drive): {
+						Pages:       pagesOf(rootAnd(), rootAnd()),
+						DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
+					},
+				},
+			},
+			limiter: newPagerLimiter(control.DefaultOptions()),
+			expect: expected{
+				counts: countTD.Expected{
+					count.TotalFoldersProcessed: 2,
+					count.PagesEnumerated:       2,
+				},
+				err:      require.NoError,
+				treeSize: 1,
+				treeContainsFolderIDs: []string{
+					rootID,
+				},
+				treeContainsTombstoneIDs: []string{},
+			},
+		},
+		{
+			name: "one",
+			tree: newFolderyMcFolderFace(nil),
+			enumerator: mock.EnumerateItemsDeltaByDrive{
+				DrivePagers: map[string]*mock.DriveItemsDeltaPager{
+					id(drive): {
+						Pages:       pagesOf(rootAnd(driveItem(id(folder), name(folder), parent(0), rootID, isFolder))),
```

Comment:
Punting: I've got a big refactor to these constructors coming up later.  I don't address the drive IDs in that specifically, but it does remove a lot of the `parent(...)` chaff.  Happy to look at improving how we reference drives sometime after that, because I agree the current version isn't entirely obvious.